src/mmw/js/src/core/views.js

```diff
@@ -238,9 +240,72 @@ function getLatLngs(boundsOrShape) {
     throw 'Unable to extract latlngs from boundsOrShape argument';
 }
 
+var DeleteModal = Marionette.ItemView.extend({
+    el: '#modal-basic-target',
+
+    viewOptions: ['objToDelete'],
+
+    initialize: function(options) {
+        var self = this;
+        this.mergeOptions(options, this.viewOptions);
+        this.$el.on('hide.bs.modal', function(e) {
+            self.cleanup();
+        });
+    },
+
+    template: modalDeleteTmpl,
+
+    ui: {
+        deleteConfirmed: '.delete',
+    },
+
+    events: {
+        'click @ui.deleteConfirmed': 'deleteModel',
+    },
+
+    deleteModel: function() {
+        this.objToDelete.destroy();
```

Comment:
Still think that if this modal view is doing the actual destroying, you should pass the return of `destroy` so caller can handle success/failure.

http://backbonejs.org/#Model-destroy